### PR TITLE
Fix pending issues

### PR DESCRIPTION
**Pull request by KaviyaPeriyasamy**
This PR helps to fix the below issues:
1. Validate the mappings table in the respective process doctypes.
2. Change field type to float in dc return items table.
3. Display rejected item in print formats only if exist.
4. remove the blank line or hyphen if secondary qty not exist.
5. Handle deliver later in dc print format.

Screenshot from 2020-07-14 12-17-11

Screenshot from 2020-07-13 14-04-49

### CODE REVIEW

File: apparelo/apparelo/utils/utils.py [hunk 1]
```diff
@@ -295,3 +309,39 @@ def check_if_same_value_dict_list(dict_list, key):
 
 # things to figureout
 	# - how to separate out different tables for different size sets
+
+def validate_mappings(additional_parts, additional_parts_sizes, additional_parts_colours):
```
Comment (karthikeyan5) [inline apparelo/apparelo/utils/utils.py]: - [x] can you name the function specifically.... like "validate_additional_parts_mapping"?
- [ ] add validations where parts are added in size/colour mapping table but not added in the additional parts table (also similarly for cutting stitching)

File: apparelo/apparelo/doctype/bleaching/bleaching.py [hunk 2]
```diff
@@ -16,6 +16,18 @@
...
+	def validate_mappings(self):
+		for colour_shade_mapping in self.colour_shade_mapping:
+			found = False
+			for colours in self.types:
+				if colours.colour == colour_shade_mapping.colour:
+					found = True
+			if found:
+				continue 
+			else:
+				frappe.throw(_(f'The colour {colour_shade_mapping.colour} entered in colour shade mapping table was not found in colours table.'))
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/bleaching/bleaching.py]: remove this validation as discussed...

File: apparelo/apparelo/doctype/dyeing/dyeing.py [hunk 3]
```diff
@@ -17,6 +17,18 @@ class Dyeing(Document):
...
+	def validate_mappings(self):
+		for colour_shade_mapping in self.colour_shade_mapping:
+			found = False
+			for colours in self.colours:
+				if colours.colour == colour_shade_mapping.colour:
+					found = True
+			if found:
+				continue 
+			else:
+				frappe.throw(_(f'The colour {colour_shade_mapping.colour} entered in colour shade mapping table was not found in colours table.'))
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/dyeing/dyeing.py]: remove this validation as discussed....

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]

File: apparelo/apparelo/doctype/dyeing/dyeing.py (hunk 3 above)
Comment (KaviyaPeriyasamy) [reply apparelo/apparelo/doctype/dyeing/dyeing.py]: The above-requested changes are updated in this commit https://github.com/aerele/apparelo/pull/187/commits/09c97c4f02068b810d169d4c82f9ebfa16d1a42a

Comment (KaviyaPeriyasamy) [reply apparelo/apparelo/doctype/dyeing/dyeing.py]: Here, I updated the code to handle location-wise deliver later items for print format https://github.com/aerele/apparelo/pull/187/commits/03c7d735eb0ae63e15d79969331eed31f5ee0aeb

**Review by karthikeyan5 — APPROVED** [review]

**Pull request merged**